Lib/shelve.py

```diff
@@ -91,6 +96,15 @@ def __init__(self, dict, protocol=None, writeback=False,
         self.cache = {}
         self.keyencoding = keyencoding
 
+        if serializer is None and deserializer is None:
+            self.serializer = dumps
+            self.deserializer = loads
+        elif (serializer is None) ^ (deserializer is None):
```

Comment:
I think this should use a logical operator, not a bitwise numerical operator.

Reply:
See: https://github.com/python/cpython/pull/99632#discussion_r1050767564